### ISSUE TITLE
Mfi license

### ISSUE THREAD

**Issue by butaikis** [espressif/esp-homekit-sdk#137]
Hello. Tell me how can I get an MFi license for the esp-homekit-sdk? I left a request on the espressif website a couple of months ago, but they still haven't answered me.

**Comment by shahpiyushv** [comment]
@butaikis , are you already an [Apple MFi licensee](https://mfi.apple.com/en/home)? Had you provided your MFi license number in the request?

**Comment by butaikis** [comment]
@shahpiyushv No. I asked a general question, what do you need from me to get a license

**Comment by shahpiyushv** [comment]
MFI license is provided by Apple. You can enroll from [here](https://mfi.apple.com/en/enroll). Once you get it from Apple, share your MFi license id with our sales team. They will validate the license number on the MFi portal and then grant you the access to our MFI compliant HomeKit SDK.

The requirements for MFi licensees are governed by Apple. So, it will be better to check with them directly.

**Comment by itatboy** [comment]
@shahpiyushv 

Hello, I have always had a question. What is the difference between ESP Homekit SDK MFI and ESP Homekit SDK? Will accessories developed based on SDK MFI still prompt for unauthenticated devices when added to the Home App?

**Comment by shahpiyushv** [comment]
Just using the MFi variant of the SDK is not sufficient. You need to either integrate the MFi auth chip into the hardware or flash software tokens provided by Apple. Thereafter, the iOS will not show the prompt for "Unauthenticated accessory". You need to be an MFi licensee to get access to the MFi variant of the SDK and also to procure the MFi auth chip or software authentication tokens.

**Comment by butaikis** [comment]
Hello @shahpiyushv. 
1. Can I learn more about software tokens ? How do they work, and is there any time limit on their use, do I need to update/get the SDK from Espressif for free?

2. If I have an MFi license from Apple, do I still need to make additional payments in the future?

**Comment by shahpiyushv** [comment]
> Hello @shahpiyushv.
> 
> 1. Can I learn more about software tokens ? How do they work, and is there any time limit on their use, do I need to update/get the SDK from Espressif for free?

Software tokens are provided by Apple along with UUIDs, one pair for each accessory, which is generally programmed during manufacturing. This token is verified by the iOS controller during pairing. If the token is valid and pairing succeeds, it sends a new token to the accessory for subsequent use. There is no time based validity of the tokens (at least from accessory's perspective, not sure how Apple manages this). However, a token is valid only for one pairing attempt. 

Espressif does not charge for the SDK. You need to be an MFi licensee though.

> 2. If I have an MFi license from Apple, do I still need to make additional payments in the future Espressif?

As mentioned above, we just verify if you are a valid MFi Licensee. There is no additional payment for getting access to the MFI variant of the SDK.

> 3. What will happen to the released products if suddenly the company does not renew the MFi license from Apple?

I am not sure what happens in that case. You would have to ask Apple MFi representatives for that.

**Comment by butaikis** [comment]
@shahpiyushv Thanks for the clarifications. 
1. If we have already released a smart home controller (in HAP, the accessory is **Bridge**) and update the software, it becomes possible to use HomeKit in our controllers, then how do we download tokens?

2. What happens if we use the _homekit-sdk MFi_ but don't download the tokens?

3. One controller == unique token + UUID ?
